### ISSUE TITLE
[Magento2] Magicento conflicts with composer plugin setting "Add packages as libraries"

### ISSUE THREAD

**Issue by Koc** [enriquepiatti/Magicento#254]
I've configured composer plugin:
![default](https://user-images.githubusercontent.com/191082/28249127-c61505a0-6a58-11e7-8a05-cf97391de2f1.png)

after this all vendor folders added to ignore list
![default](https://user-images.githubusercontent.com/191082/28249134-d95bb24e-6a58-11e7-9437-d1e6f319c5de.png)

it is normal case. In fact they are properly indexed by phpstorm (for example Symfony2 plugin works ok with same composer settings). But there is no item create plugin in menus:

`alt+enter` does nothing.

`alt+ins`
![default](https://user-images.githubusercontent.com/191082/28249091-2b048e1e-6a58-11e7-915e-82efdb6ccbf5.png)

`alt+m`
![default](https://user-images.githubusercontent.com/191082/28249095-39cd0282-6a58-11e7-981f-b588fb585b49.png)

There is no any final method in class.

When I've removed `module-catalog` folder from ignored path - menu item returns on `alt+enter`.

1. Can we unify Magicento actions? There is 3 different hotkey combinations, does we need all of them? Why not reuse `alt+ins` for plugins, for example?
2. Can you ask @Haehnchen how to properly index vendors when "Add packages as libraries" setting is turned on?

**Comment by Haehnchen** [comment]
libraries are inside `allScope`

```
GlobalSearchScope.getScopeRestrictedByFileTypes(GlobalSearchScope.allScope(project), TwigFileType.INSTANCE))
GlobalSearchScope.allScope(project)
FileBasedIndex.getInstance().getContainingFiles(EventAnnotationStubIndex.KEY, eventName, GlobalSearchScope.allScope(project))
```

**Comment by Koc** [comment]
thanx @Haehnchen, you are the best as always 👍 

More info about "Add packages as libraries" setting
https://www.jetbrains.com/help/phpstorm/using-composer-dependency-manager.html

**Comment by Koc** [comment]
@enriquepiatti please fix

**Comment by enriquepiatti** [comment]
@Koc this is actually a problem with "Intentions" not with Libraries, Magicento2 is searching on libraries too (there is also a new setting in v1.7.0 to control this on Magicento2 settings).
But this should be solved now (since v1.7.0) because Intentios are now also available as Magicento2 actions (ALT+M), so instead of using the intention shortcut (ALT+ENTER) you should use ALT+M in this case (both things are doing the same anyway).
